core/src/main/scala/com/databricks/labs/remorph/intermediate/trees.scala

```diff
@@ -252,31 +265,12 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     if (ctors.isEmpty) {
       sys.error(s"No valid constructor for $nodeName")
     }
-    val allArgs: Array[AnyRef] = if (otherCopyArgs.isEmpty) {
-      newArgs
-    } else {
-      newArgs ++ otherCopyArgs
-    }
-    val defaultCtor = ctors
-      .find { ctor =>
-        if (ctor.getParameterTypes.length != allArgs.length) {
-          false
-        } else if (allArgs.contains(null)) {
-          // if there is a `null`, we can't figure out the class, therefore we should just fallback
-          // to older heuristic
-          false
-        } else {
-          val argsArray: Array[Class[_]] = allArgs.map(_.getClass)
-          isAssignable(argsArray, ctor.getParameterTypes)
-        }
-      }
-      .getOrElse(ctors.maxBy(_.getParameterTypes.length)) // fall back to older heuristic
+
+    val (defaultCtor, allArgs) = locateConstructor(ctors, newArgs, appendOrigin = false)
```

Comment:
the constructor may or not require the Origin paranter from the 2nd parameter list, so try locating constructor with and without it